ang/test/civicase-base/services/case-type-filterer/case-type-filterer.service.spec.js

```diff
@@ -97,5 +108,56 @@
         });
       });
     });
+
+    describe('active and disabled case types', () => {
+      describe('when filtering without specifying the case type active field', () => {
+        beforeEach(() => {
+          expectedCaseTypes = _.filter(allCaseTypes, {
+            is_active: '1'
+          });
+
+          returnedCaseTypes = CaseTypeFilterer.filter({});
+        });
+
+        it('returns all active case types', () => {
+          expect(returnedCaseTypes)
+            .toEqual(jasmine.arrayWithExactContents(expectedCaseTypes));
```

Comment:
Do we need `arrayWithExactContents`? I thought `toEqual` already takes care of that.

Reply:
I removed `arrayWithExactContents` from most statements except one: the one where we use `sample(2)` since the arrays can come in random order, `toEqual` expects the values to come in a very specific order, but `arrayWithExactContents` does not.

Reply:
Great. But another question about this, you are already storing the sampled array inside `expectedCaseTypes`, so will it really come in a random order? because we are storing the random ordered values in that variable already.

Reply:
The sampled array will come in a random order. Let's say that the case type IDs will be `[3, 1]`. The list of case types are stored in ascending order so we'd have `[1,2,3,4]`, etc. When the case type filterer service goes through each case type to see if they match the expected value (in this case, the filter value is `{id: {IN: [3, 1] }}` it will return the case types in this order `[1, 3]` so the comparison would fail if using `.toEqual` by itself.